src/redux/books/booksSlice.js

```diff
@@ -3,19 +3,50 @@
 import { createSlice } from '@reduxjs/toolkit';
 
 const initialState = {
-  books: [],
+  booksGallery: // Initial state:
+  [
+    {
+      item_id: 'item1',
+      title: 'The Great Gatsby',
+      author: 'John Smith',
+      category: 'Fiction',
+    },
+    {
+      item_id: 'item2',
+      title: 'Anna Karenina',
+      author: 'Leo Tolstoy',
+      category: 'Fiction',
+    },
+    {
+      item_id: 'item3',
+      title: 'The Selfish Gene',
+      author: 'Richard Dawkins',
+      category: 'Nonfiction',
+    },
+  ],
 };
 
 const booksSlice = createSlice({
   name: 'books',
   initialState,
   reducers: {
     addBook: (state, actions) => {
-      state.books = [...state.books, actions.payload];
+      const itemId = 'item'.concat((state.booksGallery.length + 1).toString());
+      const book = {
+        ...actions.payload,
+        item_id: itemId,
+      };
+
+      return {
+        booksGallery: [...state.booksGallery, book],
+      };
     },
 
     removeBook: (state, actions) => {
```

Comment:
- [ ] I suggest you should use `action` instead of `actions` since the function is accepting only one action at a time :smile: